.github/scripts/build-qt.cmake

```diff
@@ -0,0 +1,349 @@
+endif()
+
+include(QtMacDeploymentTarget)
+set_mac_deployment_target(${QT_VERSION})
+
+include(BuildDependency)
+
+if(ANDROID)
+	list(APPEND BASE_FLAGS
+		-android-ndk "${ANDROID_NDK_ROOT}"
+		-android-sdk "${ANDROID_SDK_ROOT}"
+		-android-abis "${ANDROID_ABI}"
+	)
+
+	if(QT_VERSION VERSION_GREATER_EQUAL 6)
+		list(APPEND BASE_FLAGS
+			-platform android-clang
+			-qt-host-path "${ANDROID_HOST_PATH}"
+		)
+	else()
+		list(APPEND BASE_FLAGS
+			-xplatform android-clang
+			-no-warnings-are-errors
+			-disable-rpath
+			# Qt5 still uses minimum 21 but NDK 25 minimum is 23
+			-android-ndk-platform "${ANDROID_PLATFORM}"
+			-hostprefix "${CMAKE_INSTALL_PREFIX}"
+		)
+	endif()
+
+	if(ANDROID_ABI STREQUAL "armeabi-v7a")
+		set(openssl_abi arm)
+	elseif(ANDROID_ABI STREQUAL "arm64-v8a")
+		set(openssl_abi arm64)
+	elseif(ANDROID_ABI STREQUAL "x86")
+		set(openssl_abi x86)
+	elseif(ANDROID_ABI STREQUAL "x86_64")
+		set(openssl_abi x86_64)
+	else()
+		message(FATAL_ERROR "Unknown Android ABI '${ANDROID_ABI}' for OpenSSL")
+	endif()
+
+	get_android_toolchain(android_toolchain_path "${ANDROID_NDK_ROOT}")
+	list(APPEND OPENSSL_ENV
+		"ANDROID_NDK_HOME=${ANDROID_NDK_ROOT}"
+		"PATH=${android_toolchain_path}/bin:$ENV{PATH}"
+	)
+	list(APPEND OPENSSL_FLAGS android-${openssl_abi})
+	if(ANDROID_PLATFORM MATCHES "^android-([0-9]+)$")
+		list(APPEND OPENSSL_FLAGS "-D__ANDROID_API__=${CMAKE_MATCH_1}")
+	else()
+		message(FATAL_ERROR "Invalid Android platform '${ANDROID_PLATFORM}'")
+	endif()
+	list(APPEND OPENSSL_MAKE_FLAGS SHLIB_VERSION_NUMBER= SHLIB_EXT=_1_1.so)
+endif()
+
+if(WIN32)
+	set(SCRIPT_SUFFIX .bat)
+	list(APPEND BASE_FLAGS -mp -schannel)
+endif()
+
+if(UNIX AND NOT APPLE AND NOT ANDROID)
+	list(APPEND BASE_FLAGS -system-freetype -fontconfig)
+else()
+	list(APPEND BASE_FLAGS -qt-freetype)
+endif()
+
+# https://bugreports.qt.io/browse/QTBUG-72846 regressed in Qt6 due to
+# https://gitlab.kitware.com/cmake/cmake/-/issues/23864
+if(NOT APPLE OR QT_VERSION VERSION_LESS 6)
+	list(APPEND BASE_FLAGS -ltcg)
+endif()
+
+if(USE_ASAN)
+	list(APPEND BASE_FLAGS -sanitize address)
+endif()
+
+if(QT_VERSION VERSION_GREATER_EQUAL 6.4)
+	list(APPEND MULTIMEDIA_FLAGS
+		-no-feature-ffmpeg
+		-no-feature-spatialaudio
+	)
+endif()
+
+if(QT_VERSION VERSION_GREATER_EQUAL 6)
+	set(BASE_GENERATOR
+		CMAKE NO_DEFAULT_BUILD_TYPE
+		CONFIGURATOR configure${SCRIPT_SUFFIX}
+	)
+	set(MODULE_GENERATOR
+		CMAKE NO_DEFAULT_FLAGS
+		CONFIGURATOR ${CMAKE_INSTALL_PREFIX}/bin/qt-configure-module${SCRIPT_SUFFIX} @source_dir@
+	)
+
+	list(APPEND BASE_FLAGS
+		-no-feature-qmake
+	)
+
+	# "--" sends the cmake flags of the configurator to cmake instead of
+	# treating them as unknown extra defines
+	list(APPEND BASE_FLAGS --)
+
+	if(OPENSSL)
+		list(APPEND BASE_FLAGS "-DOPENSSL_ROOT_DIR=${CMAKE_INSTALL_PREFIX}")
+	endif()
+
+	set(TOOLS_FLAGS
+		-no-feature-clang
+		-no-feature-clangcpp
+	)
+else()
+	set(URL_LICENSE opensource-)
```

Comment:
This can be addressed later if it is decided that CI really needs to start building old versions because they keep being inadvertently broken. I don’t have the energy now to create another build :-)